File: src/dataset/cpp/beer-song_exercism-chrisnp/claude/ClaudeSonnet4_beer_song_v2.cpp
#include "beer_song.h"
#include <string>

std::string beer_song::verse(int n) noexcept {
    static const std::string case0 = "No more bottles of beer on the wall, "
                                    "no more bottles of beer.\nGo "
                                    "to the store and buy some more, "
                                    "99 bottles of beer on the wall.\n";
    static const std::string case1 = "1 bottle of beer on the wall, "
                                    "1 bottle of beer.\nTake it down and "
                                    "pass it around, no more bottles "
                                    "of beer on the wall.\n";
    static const std::string case2 = "2 bottles of beer on the wall, "
                                    "2 bottles of beer.\nTake one down "
                                    "and pass it around, 1 bottle "
                                    "of beer on the wall.\n";
    
    switch (n) {
        case 0: return case0;
        case 1: return case1;
        case 2: return case2;
        default:
            return std::to_string(n) + " bottles of beer on the wall, " +
                   std::to_string(n) + " bottles of beer.\n"
                   "Take one down and pass it around, " +
                   std::to_string(n - 1) + " bottles of beer on the wall.\n";
    }
}

std::string beer_song::sing(int start, int end) noexcept {
    std::string result;
    result.reserve((start - end + 1) * 120); // Approximate verse length
    
    for (int n = start; n >= end; --n) {
        if (n != start) result += '\n';
        result += verse(n);
    }
    return result;
}

std::string beer_song::sing(int start) noexcept {
    return sing(start, 0);
}

File: src/dataset/Java/beer-song_exercism-uzilan/claude/ClaudeSonnet4_BeerSong_v2.java
import static java.lang.String.format;

public class BeerSong {

    public String sing(final int from, final int count) {
        StringBuilder result = new StringBuilder();
        
        for (int i = from; i >= from - count + 1; i--) {
            if (i == 0) {
                result.append(format(
                        "No more bottles of beer on the wall, no more bottles of beer.\n" +
                        "Go to the store and buy some more, 99 bottles of beer on the wall.\n"));
            } else {
                result.append(format(
                        "%s of beer on the wall, %s of beer.\n" +
                        "Take %s down and pass it around, %s of beer on the wall.\n",
                        bottles(i), bottles(i), one(i), bottles(i - 1)));
            }
            
            if (i > from - count + 1) {
                result.append("\n");
            }
        }
        
        result.append("\n");
        return result.toString();
    }

    public String singSong() {
        return sing(99, 100);
    }

    private String bottles(int count) {
        switch (count) {
            case 0:
                return "no more bottles";
            case 1:
                return "1 bottle";
            default:
                return count + " bottles";
        }
    }

    private String one(int count) {
        return count == 1 ? "it" : "one";
    }
}